File: frontend/src/types/customer.ts
export interface Customer {
  id: number
  firstName: string
  lastName: string
  dateOfBirth: string
  createdAt: string
}

export interface CreateCustomerRequest {
  firstName: string
  lastName: string
  dateOfBirth: string
}

export interface FieldError {
  field: string
  message: string
}

export interface ApiError {
  errors: FieldError[]
}

File: backend/src/main/kotlin/com/example/customers/domain/Customer.kt
package com.example.customers.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Past
import java.time.Instant
import java.time.LocalDate

@Entity
data class Customer(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @field:NotBlank
    val firstName: String,
    @field:NotBlank
    val lastName: String,
    @field:NotNull @field:Past
    val dateOfBirth: LocalDate,
    val createdAt: Instant = Instant.now(),
)
